rtos/tests/TESTS/mbed_rtos/basic/CMakeLists.txt

```
@@ -1,18 +1,23 @@
-# Copyright (c) 2020 ARM Limited. All rights reserved.
+# Copyright (c) 2020-2021 ARM Limited. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
-cmake_minimum_required(VERSION 3.19.0 FATAL_ERROR)
+include(mbed_greentea)
 
-set(MBED_PATH ${CMAKE_CURRENT_SOURCE_DIR}/../../../../.. CACHE INTERNAL "")
-set(TEST_TARGET mbed-rtos-basic)
+if(DEFINED MBED_RTOS_SINGLE_THREAD OR MBED_GREENTEA_TEST_BAREMETAL)
```

I not sure `if(DEFINED MBED_RTOS_SINGLE_THREAD` works, because `if(DEFINED` checks if a CMake variable rather than a C/C++ macro. But `MBED_RTOS_SINGLE_THREAD` is a macro (i.e. compile definition) https://github.com/ARMmbed/mbed-os/blob/50c135292c42b69f7c87dba351af378a9e1e8f77/tools/cmake/toolchains/GCC_ARM.cmake#L48-L53

This macro is set when a "small" C lib (MicroLib or newlib-nano) is used, because it doesn't provide thread safety.

So maybe we can do
```suggestion
if(MBED_C_LIB STREQUAL "small" OR MBED_GREENTEA_TEST_BAREMETAL)
```

Same comment for the other tests that check this macro.

---

isn't just baremetal macro sufficient for this test?

---

I think we do need to check `MBED_RTOS_SINGLE_THREAD`, which is set when `"c_lib": "small"` while RTOS is enabled, to limit RTOS to a single thread due to lack of thread safety in small c libs.

I'm not sure why we didn't simply force bare metal only when a small C lib is used, but maybe there was a reason for it.

---

Hmm, it's actually weird. Searching how `MBED_RTOS_SINGLE_THREAD` is actually used: https://github.com/ARMmbed/mbed-os/search?p=1&q=MBED_RTOS_SINGLE_THREAD

It's only used by test cases + an IAR file. There's no enforcement of single thread in any source code. So it doesn't make any difference at all (apart from tests)? Maybe we should've instead get rid of this macro, and force users to use baremetal when small C lib is used?

---

Previous discussion here: #4885

---

For now maybe keeping the check is the safest?